x/marker/keeper/query_server.go

```diff
@@ -149,3 +149,13 @@ func (k Keeper) Access(c context.Context, req *types.QueryAccessRequest) (*types
 	}
 	return &types.QueryAccessResponse{Accounts: marker.GetAccessList()}, nil
 }
+
+// DenomMetadata query for metadata on denom
+func (k Keeper) DenomMetadata(c context.Context, req *types.QueryDenomMetadataRequest) (*types.QueryDenomMetadataResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	metadata := k.bankKeeper.GetDenomMetaData(ctx, req.Denom)
```

Comment:
Can we validate denom here? Always good practice to validate rpc request inputs.

Reply:
I added a check similar to one in the cosmos sdk bank module to check that the denom is set.